js/ui/tree_list/ui.tree_list.data_controller.js

```diff
@@ -93,6 +93,13 @@ export const DataController = dataControllerModule.controllers.data.inherit((fun
                 that.executeAction(isExpanded ? 'onRowCollapsing' : 'onRowExpanding', args);
 
                 if(!args.cancel) {
+                    const editingMode = this.option('editing.mode');
+                    if(editingMode === 'form' || editingMode === 'row') {
+                        this.option('editing.editRowKey', null);
+                        this.option('editing.editColumnName', null);
+                        this.option('editing.changes', []);
+                    }
+
```

Comment:
what is this code for?

Reply:
In TreeList we need to reset editing state after node expand/collapse.

Reply:
Why?